templates/essentials.yaml

```diff
@@ -429,6 +429,21 @@ Resources:
                   - "ec2:CreateTags"
                 Resource:
                   - 'arn:aws:ec2:*::snapshot/*'
+  AWSSSMManagedInstanceRole:
+    Type: 'AWS::IAM::Role'
+    Properties:
+      Path: '/'
+      AssumeRolePolicyDocument:
+        Version: '2012-10-17'
+        Statement:
+        - Effect: Allow
+          Principal:
+            Service:
+            - 'ec2.amazonaws.com'
+          Action: 
+            - 'sts:AssumeRole'
+      ManagedPolicyArns: 
+            - 'arn:aws:iam::aws:policy/AmazonSSMManagedInstanceCore'
```

Comment:
From reading the [documentation](https://docs.aws.amazon.com/systems-manager/latest/userguide/automation-cf.html), it looks like you actually want the `arn:aws:iam::aws:policy/service-role/AmazonSSMAutomationRole`  role?

Reply:
why not set this to `AmazonSSMAutomationRole`?

you will also need to add the policy from `TagRootVolumeRole` resource here as well.

Reply:
I believe the `AmazonSSMAutomationRole` is meant for resources that would make changes to instances that have the `AmazonSSMManagedInstanceCore` policy attached. Like [this example](https://docs.aws.amazon.com/systems-manager/latest/userguide/automation-walk-security-assume.html).

Reply:
took another look and you are correct.